journal/202211.md

```diff
@@ -0,0 +1,514 @@
+- Added new [IPC events](https://github.com/decred/dcrd/pull/3020) to notify the parent process of the address and port assigned to dcrd. In configurations where dcrd is controlled by another process like Decrediton or an automated test runner, that parent process needs to know which addresses and ports dcrd uses for its P2P interface (talking to the public network) and RPC interface (local communications with wallets and other clients). This change provides a reliable way to discover dcrd's ports when they are randomly assigned by the operating system and not configured by the parent process.
+- Always respond to [`getheaders`](https://github.com/decred/dcrd/pull/3030) requests from peers. Previous logic would not respond to `getheaders` if the local chain was not considered to be fully synced with the network. In the old "blocks first" syncing model it helped to avoid some undesirable corner cases, like unnecessarily downloading blocks. It had downsides too, like a node could appear to be unresponsive or stalled to its peers, while being active and simply not having any interesting data to share. In the current "headers first" syncing model this limitation is no longer necessary, and serving up headers before the peer is known to be current no longer leads to undesirable behavior.
+- Other internal/infra changes { 3021 3023 3022 - drop after review }
+
+RPC server:
+
+- Optimized WebSocket client disconnection handling and fixed spurious messages when WebSocket clients disconnect.
+- Modified the [lifecycle](https://github.com/decred/dcrd/pull/3025) of WebSocket clients to use contexts, which is more flexible for the calling code.
+- Added [cancellation support](https://github.com/decred/dcrd/pull/3027) for the `getwork` request (used by miners) and optimized handling of multiple concurrent `getwork` requests.
+  - { _could not understand deeper what blocked what before this cange, or what setups experienced a bottleneck. maybe current text is enough but I can add more if it's fun_ }
+- RPC testing framework has been extracted into the new `dcrtest/dcrdtest` package in its own repository. It is used to write integration-level tests which exercise features and behaviors of the fully compiled dcrd binary (as opposed to unit tests, which test small parts of the code).
+  - { _why was it extracted?_ }
```

Comment:
It provides nicer functionality in terms of selecting a specific version of `dcrd` to test against by making use of Go's module system instead of always compiling the latest master branch.  This allows consumers of the testing framework, such as `dcrlnd`, to provide more quality assurance by testing against release versions as well as the latest master.

Reply:
Thanks, added. Good to know more repos like dcrlnd can use the tests more easily now.